esp-hal-common/src/i2c.rs

```diff
@@ -274,6 +274,20 @@ where
         scl: impl Peripheral<P = SCL> + 'd,
         frequency: HertzU32,
         clocks: &Clocks,
+    ) -> Self {
+        Self::new_with_timeout(i2c, sda, scl, frequency, clocks, None)
+    }
+
+    /// Create a new I2C instance with a custom timeout value.
+    /// This will enable the peripheral but the peripheral won't get
+    /// automatically disabled when this gets dropped.
+    pub fn new_with_timeout<SDA: OutputPin + InputPin, SCL: OutputPin + InputPin>(
+        i2c: impl Peripheral<P = T> + 'd,
+        sda: impl Peripheral<P = SDA> + 'd,
+        scl: impl Peripheral<P = SCL> + 'd,
+        frequency: HertzU32,
+        clocks: &Clocks,
+        timeout: Option<u32>,
```

Comment:
Feels a bit strange to be mixing `fugit` types with primitive types within the same API, is there a particular reason `u32` was chosen here instead of say `DurationU32`?

Reply:
The timeout is not expressed in time but in 2 ^ timeout clock cycles ... while it would be possible trying to do a lot of math to have a duration here it's probably not really worth it

Reply:
But maybe we should do it anyways - or document it at least